lib/cc/analyzer/path_patterns.rb

```diff
@@ -0,0 +1,26 @@
+module CC
+  module Analyzer
+    class PathPatterns
+      def initialize(patterns, root = Dir.pwd)
+        @patterns = patterns
+        @root = root
+      end
+
+      def expanded
+        @expanded ||= expand
+      end
+
+      private
+
+      def expand
+        results = Dir.chdir(@root) do
+          @patterns.map do |pattern|
+            Dir.glob(pattern)
+          end
+        end
+
+        results.flatten.sort.uniq
```

Comment:
Someone should add a rubocop check for flat_map :)



—Bryan Helmkamp, Founder, Code Climate

bryan@brynary.com / 646-379-1810 / @brynary




Typed with thumbs.

On Thu, Jul 23, 2015 at 6:31 PM, pat brisbin <notifications@github.com>
wrote:

>> +      end
>> +
>> +      def expanded
>> +        @expanded ||= expand
>> +      end
>> +
>> +      private
>> +
>> +      def expand
>> +        results = Dir.chdir(@root) do
>> +          @patterns.map do |pattern|
>> +            Dir.glob(pattern)
>> +          end
>> +        end
>> +
>> +        results.flatten.sort.uniq
> Could remove `flatten` by using `flat_map` above.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/codeclimate/codeclimate/pull/67/files#r35389255